### PR TITLE
add test demonstrating zero-diff check failure

### PR DESCRIPTION
**Pull request by deads2k**
`make && ./openshift-tests run-test "[sig-apimachinery] server-side-apply zero diff detection should not update when the existing values have not changed [Suite:openshift/conformance/parallel]"`

This is currently failing with 
```
   STEP: Destroying namespace "e2e-test-server-side-apply-zero-diff-hzpgj" for this suite. @ 09/12/24 12:41:59.485
  • [FAILED] [4.844 seconds]
  [sig-apimachinery] server-side-apply zero diff detection [It] should not update when the existing values have not changed [Suite:openshift/conformance/parallel]
  github.com/openshift/origin/test/extended/operators/server_side_apply_zero_diff.go:23

    [FAILED] Expected
        <string>: modification done
    to equal
        <string>: nothing to apply
    In [It] at: github.com/openshift/origin/test/extended/operators/server_side_apply_zero_diff.go:65 @ 09/12/24 12:41:58.761
  ------------------------------

  Summarizing 1 Failure:
    [FAIL] [sig-apimachinery] server-side-apply zero diff detection [It] should not update when the existing values have not changed [Suite:openshift/conformance/parallel]
    github.com/openshift/origin/test/extended/operators/server_side_apply_zero_diff.go:65

  Ran 1 of 1 Specs in 4.844 seconds
  FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
```

### CODE REVIEW

File: test/extended/operators/server_side_apply_client_copy.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	}
+	return &ret.Spec.OperatorSpecApplyConfiguration, nil
+}
+
+func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorStatusApplyConfiguration, error) {
+	castObj := &operatorv1.OpenShiftAPIServer{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to OpenShiftAPIServer: %w", err)
+	}
+	ret, err := applyoperatorv1.ExtractOpenShiftAPIServerStatus(castObj, fieldManager)
```
Comment (deads2k) [inline test/extended/operators/server_side_apply_client_copy.go:189]: I messed up this name.

Comment (p0lyn0mial) [reply test/extended/operators/server_side_apply_client_copy.go:189]: ah, so instead of extracting the status you were extracting the spec thus the diff!
